src/main/kotlin/io/github/anki/anki/controller/CardsController.kt

```diff
@@ -8,52 +8,52 @@ import io.github.anki.anki.controller.dto.mapper.toCard
 import io.github.anki.anki.controller.dto.mapper.toDto
 import io.github.anki.anki.controller.dto.mapper.toPagination
 import io.github.anki.anki.service.CardsService
+import io.github.anki.anki.service.model.Card
 import io.github.anki.anki.service.secure.SecurityService
 import jakarta.validation.Valid
 import org.slf4j.LoggerFactory
-import org.springframework.http.HttpHeaders
 import org.springframework.http.HttpStatus
 import org.springframework.web.bind.annotation.DeleteMapping
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PatchMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RequestBody
-import org.springframework.web.bind.annotation.RequestHeader
 import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestController
+import reactor.core.publisher.Mono
 
 @RestController
 @RequestMapping(CardsController.BASE_URL)
 class CardsController(
     private val cardService: CardsService,
-    val securityService: SecurityService,
+    private val securityService: SecurityService,
 ) {
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
     fun createCard(
         @Valid @RequestBody request: NewCardRequest,
         @PathVariable deckId: String,
-        @RequestHeader header: HttpHeaders,
-    ): CardDtoResponse {
-        LOG.info("IN: $CardsController ${BASE_URL} create card in deck with id = $deckId")
-        val card =
-            cardService.createNewCard(
-                userId = securityService.jwtUtils.getUserIdFromAuthHeader(header),
-                request.toCard(deckId),
-            )
-        LOG.info("OUT: $CardsController ${BASE_URL} created card ${card.id} with id = $deckId")
-        return card.toDto()
-    }
+    ): Mono<CardDtoResponse> =
+        securityService
+            .getUserIdFromAuthentication()
+            .doFirst { LOG.info("IN: $CardsController $BASE_URL create card $request in deck with id = $deckId") }
+            .flatMap {
+                cardService.createNewCard(
+                    userId = it,
+                    request.toCard(deckId),
+                )
+            }
+            .map(Card::toDto)
+            .doOnNext { LOG.info("OUT: $CardsController $BASE_URL created card $it with id = $deckId") }
```

Comment:
inlining variables into log message is bad practice.

```suggestion
            .doOnNext { LOG.info("OUT: $CardsController $BASE_URL created card $it with id = {}", deckId) }
```